cpp/src/arrow/dataset/projector.cc

```diff
@@ -63,7 +63,7 @@ Status RecordBatchProjector::SetDefaultValue(FieldRef ref,
 
 Result<std::shared_ptr<RecordBatch>> RecordBatchProjector::Project(
     const RecordBatch& batch, MemoryPool* pool) {
-  if (from_ == nullptr || !batch.schema()->Equals(*from_)) {
+  if (from_ == nullptr || !batch.schema()->Equals(*from_, false /*check_metadata*/)) {
```

Comment:
Nit: we should try to use `/*param=*/` consistently (this is used elsewhere)

Reply:
Ok!